### PR TITLE
🧪 Test isCitationFormat

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** Added tests for the `isCitationFormat` function in `apps/api/src/utils/citation.ts` which lacked test coverage.
📊 **Coverage:** Covered both valid (`bibtex`, `biblatex`, `apa`, `ieee`, `mla`, `plain`) and invalid citation formats (e.g., `chicago`, `BIBTEX`, `""`).
✨ **Result:** Test coverage for `isCitationFormat` increased to 100%.

---
*PR created automatically by Jules for task [4465185451225415834](https://jules.google.com/task/4465185451225415834) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

この PR は `isCitationFormat` 関数（`apps/api/src/utils/citation.ts`）のテストが存在しなかった部分に対し、Vitest を使ったユニットテストを追加するものです。

- 有効なフォーマット 6 種類（`bibtex`, `biblatex`, `apa`, `ieee`, `mla`, `plain`）を `it.each` で個別テストケースとして検証し、デバッグ時の可視性を確保している。
- 無効なフォーマット（大文字・余白付き・空文字・非文字列型の `null` / `undefined` / `123` など）も網羅しており、`satisfies unknown[]` + `format as string` の TypeScript パターンを使って型安全に非文字列値のランタイム動作を検証している。
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

テストのみの変更でプロダクションコードへの影響はなく、安全にマージできる。

変更はテストファイル 1 つに限定されており、既存の `isCitationFormat` 実装と正確に対応した有効・無効ケースを網羅している。ロジック上の欠陥や型の不整合は確認されなかった。

特に注意が必要なファイルはない。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/api/src/utils/__tests__/citation.test.ts | `isCitationFormat` のテストを追加。有効 6 種・無効 8 種を `it.each` で網羅しており、論理的・型的な問題は見当たらない。 |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["describe: isCitationFormat"] --> B["it.each(validFormats)\n有効フォーマット 6 種"]
    A --> C["it.each(invalidFormats)\n無効値 8 種"]
    B --> B1["bibtex → true"]
    B --> B2["biblatex → true"]
    B --> B3["apa → true"]
    B --> B4["ieee → true"]
    B --> B5["mla → true"]
    B --> B6["plain → true"]
    C --> C1["chicago → false"]
    C --> C2["BIBTEX → false"]
    C --> C3["空文字 → false"]
    C --> C4["unknown → false"]
    C --> C5["bibtex（末尾スペース）→ false"]
    C --> C6["null → false"]
    C --> C7["undefined → false"]
    C --> C8["123 → false"]
```

<sub>Reviews (4): Last reviewed commit: ["Test citation format validation"](https://github.com/hiroki-org/openshelf/commit/950f8622f66ddc6566f594206dce1f7deadbbf0b) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=32528636)</sub>

### CODE REVIEW

**Comment by is0692vs** [comment]
@greptile review

File: apps/api/src/utils/__tests__/citation.test.ts [hunk 1]
```diff
@@ -170,3 +170,19 @@ describe("buildCitation", () => {
...
+            expect(isCitationFormat(format)).toBe(true);
+        });
+    });
+
+    it("returns false for invalid citation formats", () => {
+        const invalidFormats = ["chicago", "BIBTEX", "", "unknown", "bibtex "];
+        invalidFormats.forEach((format) => {
+            expect(isCitationFormat(format)).toBe(false);
+        });
+    });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/utils/__tests__/citation.test.ts]: 対応しました。現在のテストは it.each で有効/無効ケースを分離し、null/undefined/数値などの非文字列入力も isCitationFormat 側の unknown ガードで検証しています。

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
@greptile review

**Comment by is0692vs** [comment]
I've added the missing tests that dropped coverage during the refactor.  missing initials and default parameter cases are all covered and coverage is fully restored to 100%!

**Comment by is0692vs** [comment]
I've added the missing tests that dropped coverage during the refactor. toApaAuthorName missing initials and default parameter cases are all covered and coverage is fully restored to 100%!

**Comment by is0692vs** [comment]
I've added the missing tests for `buildCitation` that dropped coverage during the refactor. This includes edge cases for `toApaAuthorName` without initials and various default parameter/falsy cases. The coverage is now fully restored and improved.
